src/functional/ffront/decorator.py

```diff
@@ -504,7 +508,8 @@ def __gt_itir__(self) -> itir.FunctionDefinition:
         if hasattr(self, "__cached_itir"):
             return getattr(self, "__cached_itir")  # noqa: B009
 
-        itir_node: itir.FunctionDefinition = FieldOperatorLowering.apply(self.foast_node)
+        itir_node = FieldOperatorLowering.apply(self.foast_node)
+        assert isinstance(itir_node, itir.FunctionDefinition)
```

Comment:
maybe we can fix the return type annotation of `FieldOperatorLowering.apply` to make this work without assert?

Reply:
I am not sure that the return type annotation is wrong. `FieldOperatorLowering.apply` is also used in foast_to_itir.py/_visit_shift, where I believe that itir.Expr is correct

Reply:
I believe it could be done.

By changing the type annotations of `FieldOperatorLowering.apply`  like so:

```python
    ...
    def apply(self, node: foast.FunctionDefinition) -> itir.FunctionDefinition:
        ...
```

and changing the occurrence in `InsideReductionLowering._visit_shift` to

```python
        self.lambda_params[uid] = FieldOperatorLowering().visit(node)
```

we should get the typing we want in both places.